jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/datagrip/auth/SecretsManagerAuth.kt

```diff
@@ -52,9 +56,26 @@ class SecretsManagerAuth : DatabaseAuthProvider, CoroutineScope by ApplicationTh
     ): CompletionStage<ProtoConnection>? {
         LOG.info { "Intercepting db connection [$connection]" }
         return future {
-            val connectionSettings = getConfiguration(connection)
-            val credentials = getCredentials(connection.runConfiguration.project, connectionSettings)
-            DatabaseCredentialsAuthProvider.applyCredentials(connection, credentials, true)
+            var result = Result.Succeeded
+            val project = connection.runConfiguration.project
+            try {
+                val connectionSettings = getConfiguration(connection)
+                val credentials = getCredentials(connection.runConfiguration.project, connectionSettings)
+                DatabaseCredentialsAuthProvider.applyCredentials(connection, credentials, true)
+            } catch (e: Throwable) {
+                result = Result.Failed
+                throw e
+            } finally {
+                // We don't have access to service information here. What we do have access to is the database driver
+                // which matches up with the engine field for mysql, postgres, and redshift which is what we currently
+                // support. TODO find a more direct way to do this
+                val driver = connection.connectionPoint.databaseDriver.id
+                if (driver == "redshift") {
+                    RedshiftTelemetry.getCredentials(project, result, SecretsManager)
+                } else {
```

Comment:
nit: why not explicit RDS?

Reply:
We can expand secretsmanager auth to other database engines easily, so less places to change